draft-ietf-taps-interface.md

```diff
@@ -1780,38 +1772,170 @@ that the other endpoint is done sending on a connection.
 
 Any calls to Receive once the Final Message has been delivered will result in errors.
 
-## Receiver-side De-framing over Stream Protocols {#receive-framing}
+# Message Framers {#framing}
+
+Message Framers are implementations of simple protocols that can convert
```

Comment:
A protocol usually includes more than just adding and parsing headers: it also defines a sequence of communication (and that may come with timeouts and such). I think framers can be simpler than that - if a framer just adds and removes a header, I think it's wrong to call it a protocol. You talk about simple application protocols as framers by example later in the text, and I think that's okay - I just don't like the statement here that says that they "are implementations of simple protocols". Also, there's no need for them to be simple... I also wouldn't want to claim that. Can we just rephrase this to something like "Message Framers are converters that translate between application Message data and raw transport protocol data."  ?

Reply:
They are one of the simplest forms of protocol, I'd argue. They certainly can have flags, etc, in their headers that do some communication. For example, WebSocket is a protocol and certainly is something you'd implement as a framer.